bundles/org.openhab.binding.icalendar/src/main/resources/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,103 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="icalendar"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:thing="https://openhab.org/schemas/thing-description/v1.0.0"
+	xsi:schemaLocation="https://openhab.org/schemas/thing-description/v1.0.0 https://openhab.org/schemas/thing-description-1.0.0.xsd">
+
+	<thing-type id="calendar">
+		<label>Calendar</label>
+		<description>Calendar based on an iCal calendar.</description>
+
+		<channels>
+			<channel id="current_title" typeId="event_current_title" />
+			<channel id="current_start" typeId="event_current_start" />
+			<channel id="current_end" typeId="event_current_end" />
+			<channel id="current_presence" typeId="event_current_presence" />
+			<channel id="next_title" typeId="event_next_title" />
+			<channel id="next_start" typeId="event_next_start" />
+			<channel id="next_end" typeId="event_next_end" />
+		</channels>
+
+		<config-description>
+			<parameter-group name="source">
+				<label>Source Settings</label>
+			</parameter-group>
+			<parameter-group name="parsing">
+				<label>Parsing Settings</label>
+			</parameter-group>
+			<parameter name="url" type="text" required="true" groupName="source">
+				<label>URL</label>
+				<description>URL for downloading iCalendar events</description>
+				<context>url</context>
+			</parameter>
+			<parameter name="refreshTime" type="integer" required="true" min="1" unit="min" groupName="source">
+				<label>Refresh Time</label>
+				<description>Frequency to scan for changes in minutes</description>
+				<unitLabel>minutes</unitLabel>
```

Comment:
Not needed if you use the `unit` attribute.

Reply:
OK.